addons/im_livechat/static/src/messaging/entity/thread/thread.js

```diff
@@ -1,22 +1,57 @@
 odoo.define('im_livechat.messaging.entity.Thread', function (require) {
 'use strict';
 
-const { registerClassPatchEntity } = require('mail.messaging.entity.core');
+const {
+    fields: {
+        attr,
+    },
+    registerClassPatchEntity,
+    registerFieldPatchEntity,
+    registerInstancePatchEntity,
+} = require('mail.messaging.entity.core');
 
 registerClassPatchEntity('Thread', 'im_livechat.messaging.entity.Thread', {
 
-    //--------------------------------------------------------------------------
+    //----------------------------------------------------------------------
     // Public
-    //--------------------------------------------------------------------------
+    //----------------------------------------------------------------------
 
     /**
-     * @static
-     * @returns {mail.messaging.entity.Thread[]}
+     * @override
      */
-    allOrderedAndPinnedLivechats() {
-        return this.allPinnedChannels.filter(channel => channel.channel_type === 'livechat');
-    }
+    convertData(data) {
+        const data2 = this._super(data);
+        if ('correspondent_name' in data) {
+            data2.correspondent_name = data.correspondent_name;
+        }
+        return data2;
+    },
+});
+
+registerInstancePatchEntity('Thread', 'im_livechat.messaging.entity.Thread', {
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeDisplayName() {
+        if (this.channel_type === 'livechat') {
+            return this.correspondent_name;
+        }
+        return this._super();
+    },
+});
 
+registerFieldPatchEntity('Thread', 'im_livechat.messaging.entity.Thread', {
+    displayName: attr({
+        dependencies: [
```

Comment:
Maybe rename key when patching to explicit it is "adding to dependencies" and not "replacing".

Reply:
No need to support removal however, it's just a name change.